lib/miq_automation_engine/service_models/miq_ae_service_service_template_transformation_plan_task.rb

```diff
@@ -27,5 +27,15 @@ def transformation_destination(source_obj)
         wrap_results(@object.transformation_destination(source_obj.object_class.find(source_obj.id)))
       end
     end
+
+    def conversion_host=(conversion_host)
+      raise ArgumentError, "conversion_host must be nil or a MiqAeServiceConversionHost" unless conversion_host.nil? || conversion_host.kind_of?(MiqAeMethodService::MiqAeServiceConversionHost)
+
+      ar_method do
+        @object.conversion_host = conversion_host && conversion_host.instance_variable_get("@object")
```

Comment:
@fdupont-redhat Do we need this check since you are doing the same check in raise ArgumentError?

Reply:
We should avoid accessing the internal properties of the service model.  In the past I have used the fact that the service model exposes the same ID as the active_record model did a `find` on that object.

Like this: https://github.com/ManageIQ/manageiq-automation_engine/blob/master/lib/miq_automation_engine/service_models/miq_ae_service_vm.rb#L9